### ISSUE TITLE
Cannot override default error page with Laravel 11 Bootstrap Exception Handling

### ISSUE THREAD

**Issue by rndb8418** [laravel/framework#52469]
### Laravel Version

11.20.0

### PHP Version

8.3.1

### Description

In Laravel 10, it was possible to utilise Laravel's exception handling for internal/application errors, whilst having a fallback for errors that did not meet that criteria (e.g. incorrect database credentials).

For example, in Laravel 10, the `App\Exceptions\Handler@render` method could be...

```php
public function render($request, Throwable $exception)
{
    // If local environment, or an exception that should not be reported,
    // allow Laravel to handle rendering.
    if (app()->environment('local') || $this->shouldntReport($exception)) {
        return parent::render($request, $exception);
    }

    // Specific handling of TokenMismatchException (e.g., redirect with error message, or return JSON error if AJAX).
    // This CAN be handled in Laravel 11
    if ($exception instanceof TokenMismatchException) {
        return $this->handleTokenMismatch($request);
    }

    // Get an appropriate status code for the exception, defaulting to 500.
    $code = $this->getStatusCode($exception);

    // Return a generic error page to the user
    return response()->view('errors.generic', ['code' => $code], $code);
}
```

In Laravel 11, this does not appear to be possible without overriding the whole exception handler due to the loss of access to knowing if an exception should be ignored or not.  Something like this does not work as it would return the generic error pages for errors like unauthenticated etc.

```php
$exceptions->render(function (Throwable $e, Request $request) use ($exceptions) {
    // Not possible to check if an exception is unreportable
    if (app()->environment('local')) {
        return;
    }

    // Get the status code from the exception
    $code = ...;

    return response()->view('errors.generic', ['code' => $code], $code);
});
```

If no rendering is specified and it's left to Laravel, then errors that do not return specific HTTP exceptions (the easiest to replicate being incorrect database credentials) will return a generic Symfony error page. These exceptions ignore the `5xx`, `500` etc. custom error pages that you can configure.

### Steps To Reproduce

1. Generate an exception that does not have HTTP error handling, such as incorrect database credentials in the `.env`
2. Load page that triggers the exception
3. You'll see Symfony's generic error page shown

![image](https://github.com/user-attachments/assets/7d59a85c-d8ee-4d3a-9d45-e3c31091acb6)

**Comment by crynobone** [comment]
Hey there, thanks for reporting this issue.

We'll need more info and/or code to debug this further. Can you please create a repository with the command below, commit the code that reproduces the issue *as one separate commit* on the main/master branch and share the repository here?

Please make sure that you have the [latest version of the Laravel installer](https://github.com/laravel/installer) in order to run this command. Please also make sure you have both Git & [the GitHub CLI tool](https://cli.github.com/) properly set up.

    laravel new bug-report --github="--public"

Do not amend and create a separate commit with your custom changes. After you've posted the repository, we'll try to reproduce the issue.

Thanks!

**Comment by rndb8418** [comment]
After attempting to create a replicate (and being unable to), I narrowed it down to an exception occuring within our error template under certain conditions (e.g. no database/invalid credentials). This was causing by a view composer that was applied to the top-most layout, that all other layouts extended.

When Laravel attempted to render our error pages, it would encounter an exception and result in Laravel returning a Symfony error page.

I've fixed it by removing the specific logic from the top-most layout, alternatively I could have used a simplified error layout.

Apologies for the false report.
